src/main/java/com/epam/ta/reportportal/core/launch/impl/FinishLaunchHandler.java

```diff
@@ -113,10 +113,10 @@ public LaunchWithLinkRS finishLaunch(Long launchId, FinishExecutionRQ finishLaun
 		messageBus.publishActivity(event);
 		eventPublisher.publishEvent(event);
 
-		LaunchWithLinkRS rs = new LaunchWithLinkRS();
-		rs.setId(launchId);
-		rs.setNumber(launch.getNumber());
-		return rs;
+		FinishLaunchRS response = new FinishLaunchRS();
```

Comment:
Move the method of generating link somewhere into a util class. It shouldn't be in controller.
And logic of generation move to the service layer.